src/server/middleware/blob/head_blob.go

```diff
@@ -0,0 +1,56 @@
+package blob
+
+import (
+	"fmt"
+	"github.com/goharbor/harbor/src/controller/blob"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	lib_http "github.com/goharbor/harbor/src/lib/http"
+	"github.com/goharbor/harbor/src/lib/log"
+	blob_models "github.com/goharbor/harbor/src/pkg/blob/models"
+	"github.com/goharbor/harbor/src/server/middleware"
+	"net/http"
+)
+
+// HeadBlobMiddleware intercept the head blob request
+func HeadBlobMiddleware() func(http.Handler) http.Handler {
+	return middleware.New(func(rw http.ResponseWriter, req *http.Request, next http.Handler) {
+		if err := handleHead(req); err != nil {
+			lib_http.SendError(rw, err)
+			return
+		}
+		next.ServeHTTP(rw, req)
+	})
+}
+
+// handleHead ...
+func handleHead(req *http.Request) error {
+	none := lib.ArtifactInfo{}
+	art := lib.GetArtifactInfo(req.Context())
```

Comment:
rename the var or add some comment to explain this is in fact blob info?

Reply:
updated